### PR TITLE
Test mistral architecture adapter

### PR DESCRIPTION
**Pull request by indrayani21**
# Description

Adds focused unit tests for the Mistral architecture adapter in `tests/unit/model_bridge/supported_architectures/test_mistral_adapter.py`.

This PR improves test coverage for the Mistral architecture adapter by validating key architectural behaviors and configuration settings.

### What this PR includes

* Config validation for Mistral-specific settings:

  * RMS normalization
  * Rotary positional embeddings
  * Gated MLP configuration
  * Final RMS disabled
  * Attention-only disabled
  * GQA (`n_key_value_heads`) support

* Component mapping validation:

  * Embedding, attention, MLP, and normalization bridges
  * Correct HuggingFace module name mappings
  * Attention flags (`requires_attention_mask`, `requires_position_embeddings`)

* Weight conversion validation:

  * Ensures Q/K/V/O projection keys exist in conversion mappings

* Factory registration check:

  * Verifies `MistralForCausalLM` is correctly registered in `SUPPORTED_ARCHITECTURES`

### Motivation

Mistral is a key decoder-only architecture with grouped-query attention and RMSNorm-based design. This test suite ensures the architecture adapter correctly represents these structural properties and aligns with existing patterns used for Llama and CodeGen adapters.

Related to #1302

## Type of change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] This change requires a documentation update

## Checklist

* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] I have not rewritten tests affecting core interfaces or backward compatibility

### CODE REVIEW

**Comment by jlarson4** [comment]
Hi @indrayani21! As you continue to work on more of these, please start each new set of changes on `dev` or `main`. When you stack your PRs like this, it makes it difficult to manage their integration into the project.

Also, make sure to run `make check-format` & resolve any format requests on your code for each PR

File: tests/unit/model_bridge/supported_architectures/test_mistral_adapter.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        assert "blocks.{i}.attn.k.weight" in adapter.weight_processing_conversions
+        assert "blocks.{i}.attn.v.weight" in adapter.weight_processing_conversions
+        assert "blocks.{i}.attn.o.weight" in adapter.weight_processing_conversions
+
+
+# ---------------------------------------------------------------------------
+# Factory registration tests
+# ---------------------------------------------------------------------------
+
+class TestMistralFactory:
```
Comment (jlarson4) [inline tests/unit/model_bridge/supported_architectures/test_mistral_adapter.py:185]: Can we add a factory dispatch test? Currently this only checks if "MistralForCausalLM" in `SUPPORTED_ARCHITECTURES`. It never calls `ArchitectureAdapterFactory.select_architecture_adapter(cfg)` to verify dispatch actually returns a `MistralArchitectureAdapter`. You should be able to match the same test you wrote in  `test_bloom_adapter.py`

File: tests/unit/model_bridge/supported_architectures/test_mistral_adapter.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+# ---------------------------------------------------------------------------
+# GQA tests
+# ---------------------------------------------------------------------------
+
+class TestMistralGQA:
+
+    def test_n_key_value_heads(self, adapter: MistralArchitectureAdapter) -> None:
+        assert adapter.cfg.n_key_value_heads == 4
+        assert adapter.default_config["n_key_value_heads"] == 4
```
Comment (jlarson4) [inline tests/unit/model_bridge/supported_architectures/test_mistral_adapter.py:163]: The adapter unconditionally accesses `cfg.n_key_value_heads` when setting `n_key_value_heads` on `default_config`. This means a cfg without `n_key_value_heads` will raise an `AttributeError` at adapter construction. Can we add an additional test that catches this assertion?

**Comment by jlarson4** [comment]
Closing in favor of #1392
